src/conditions/base/rpc.ts

```diff
@@ -1,74 +1,32 @@
-import Joi, { Schema } from 'joi';
+import Joi from 'joi';
 
-import {
-  ETH_ADDRESS_REGEXP,
-  SUPPORTED_CHAINS,
-  USER_ADDRESS_PARAM,
-} from '../const';
-import { ContextParametersHandlerMixin } from '../context/mixin';
+import { SUPPORTED_CHAINS } from '../const';
 
-import { Condition, makeReturnValueTest } from './condition';
+import { Condition } from './condition';
+import { ethAddressOrUserAddressSchema, returnValueTestSchema } from './schema';
 
-export interface RpcConditionConfig {
-  CONDITION_TYPE: string;
-  RPC_METHODS: string[];
-  PARAMETERS_PER_METHOD: Record<string, string[]>;
-  CONTEXT_PARAMETERS_PER_METHOD: Record<string, string[]>;
-}
-
-export const RpcConditionConfig: RpcConditionConfig = {
-  CONDITION_TYPE: 'rpc',
-  RPC_METHODS: ['eth_getBalance', 'balanceOf'],
-  PARAMETERS_PER_METHOD: {
-    eth_getBalance: ['address'],
-    balanceOf: ['address'],
-  },
-  CONTEXT_PARAMETERS_PER_METHOD: {
-    eth_getBalance: [USER_ADDRESS_PARAM],
-    balanceOf: [USER_ADDRESS_PARAM],
-  },
+const rpcMethodSchemas: Record<string, Joi.Schema> = {
+  eth_getBalance: Joi.array().items(ethAddressOrUserAddressSchema).required(),
+  balanceOf: Joi.array().items(ethAddressOrUserAddressSchema).required(),
 };
 
-export const ethAddressOrUserAddress = Joi.alternatives().try(
-  Joi.string().pattern(ETH_ADDRESS_REGEXP),
-  USER_ADDRESS_PARAM
-);
-
-export const getAddressSchema = () =>
-  Joi.array().items(ethAddressOrUserAddress).required();
-
-export const rpcMethodSchema = RpcConditionConfig.RPC_METHODS.reduce(
-  (acc, method) => {
-    if (RpcConditionConfig.PARAMETERS_PER_METHOD[method].includes('address')) {
-      acc[method] = getAddressSchema();
-    } else {
-      acc[method] = Joi.array().items(Joi.string()).required();
-    }
-    return acc;
-  },
-  {} as Record<string, Schema>
-);
+const makeParameters = () =>
+  Joi.array().when('method', {
+    switch: Object.keys(rpcMethodSchemas).map((method) => ({
+      is: method,
+      then: rpcMethodSchemas[method],
+    })),
+  });
 
-export class RpcConditionBase extends Condition {
+export class RpcCondition extends Condition {
   public readonly schema = Joi.object({
     chain: Joi.number()
       .valid(...SUPPORTED_CHAINS)
       .required(),
     method: Joi.string()
-      .valid(...RpcConditionConfig.RPC_METHODS)
+      .valid(...Object.keys(rpcMethodSchemas))
       .required(),
-    parameters: Joi.when('method', {
-      switch: RpcConditionConfig.RPC_METHODS.map((method) => ({
-        is: method,
-        then: rpcMethodSchema[method],
-      })),
-    }),
-    returnValueTest: makeReturnValueTest(),
+    parameters: makeParameters(),
+    returnValueTest: returnValueTestSchema,
```

Comment:
Is the `returnValueTestSchema` itself required? I know the elements within it are, but is the specification of the `returnValuesTest` object on a condition set as required by default?

Reply:
That's a good catch, I will add `.required()` where needed.